### ISSUE TITLE
tf.keras.applications.EfficientNetB- not running as expected

### ISSUE THREAD

**Issue by mrdbourke** [tensorflow/tensorflow#42506]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Colab
- TensorFlow version (use command below): v2.3.0-0-gb36436b087 2.3.0

**Describe the current behavior**

[tf.keras.applications.EfficientNetB-](https://www.tensorflow.org/api_docs/python/tf/keras/applications/EfficientNetB0) not behaving like other tf.keras.applications modules.

In seems any of the EfficientNet modules I try to use don't function as expected as when using other modules such as [ResNet50V2](https://www.tensorflow.org/api_docs/python/tf/keras/applications/ResNet50V2) (using the same code/data setups).

Unless this is stated somewhere, I'm not sure what I'm getting wrong.

I've tried the TF Hub versions of EfficientNetB- and as feature extractors and they seem to be working fine.

**Describe the expected behavior**

EfficientNetB- modules should function like other modules in tf.keras.applications.

**Standalone code to reproduce the issue**
Running Colab notebook: https://colab.research.google.com/drive/1m3U3wREqWLCKqLhh5Gzuy_mdq5mk6h_r?usp=sharing

**Other info / logs** 

ResNetV250 output:
```Epoch 1/10
24/24 [==============================] - 17s 701ms/step - loss: 2.2448 - accuracy: 0.2240 - val_loss: 2.0786 - val_accuracy: 0.4660
Epoch 2/10
24/24 [==============================] - 16s 674ms/step - loss: 1.9990 - accuracy: 0.5800 - val_loss: 1.8562 - val_accuracy: 0.6896
Epoch 3/10
24/24 [==============================] - 16s 665ms/step - loss: 1.8456 - accuracy: 0.7000 - val_loss: 1.7827 - val_accuracy: 0.7476
Epoch 4/10
24/24 [==============================] - 16s 672ms/step - loss: 1.7713 - accuracy: 0.7667 - val_loss: 1.7561 - val_accuracy: 0.7608
Epoch 5/10
24/24 [==============================] - 16s 664ms/step - loss: 1.7313 - accuracy: 0.8053 - val_loss: 1.7457 - val_accuracy: 0.7616
Epoch 6/10
24/24 [==============================] - 16s 667ms/step - loss: 1.7049 - accuracy: 0.8187 - val_loss: 1.7340 - val_accuracy: 0.7712
Epoch 7/10
24/24 [==============================] - 16s 668ms/step - loss: 1.6779 - accuracy: 0.8347 - val_loss: 1.7246 - val_accuracy: 0.7824
Epoch 8/10
24/24 [==============================] - 16s 667ms/step - loss: 1.6577 - accuracy: 0.8520 - val_loss: 1.7222 - val_accuracy: 0.7760
Epoch 9/10
24/24 [==============================] - 16s 665ms/step - loss: 1.6476 - accuracy: 0.8747 - val_loss: 1.7189 - val_accuracy: 0.7772
Epoch 10/10
24/24 [==============================] - 16s 661ms/step - loss: 1.6372 - accuracy: 0.8853 - val_loss: 1.7163 - val_accuracy: 0.7764
```

EfficientNetB0 output (using same code and data as above, except for EfficientNetB0 not ResNet50V2):
```
Epoch 1/10
24/24 [==============================] - 18s 735ms/step - loss: 2.3056 - accuracy: 0.0787 - val_loss: 2.3026 - val_accuracy: 0.1000
Epoch 2/10
24/24 [==============================] - 16s 654ms/step - loss: 2.3048 - accuracy: 0.0827 - val_loss: 2.3026 - val_accuracy: 0.1000
Epoch 3/10
24/24 [==============================] - 16s 654ms/step - loss: 2.3048 - accuracy: 0.0907 - val_loss: 2.3026 - val_accuracy: 0.1000
Epoch 4/10
24/24 [==============================] - 15s 639ms/step - loss: 2.3051 - accuracy: 0.0867 - val_loss: 2.3027 - val_accuracy: 0.1000
Epoch 5/10
24/24 [==============================] - 15s 631ms/step - loss: 2.3052 - accuracy: 0.0773 - val_loss: 2.3026 - val_accuracy: 0.1000
Epoch 6/10
24/24 [==============================] - 15s 632ms/step - loss: 2.3052 - accuracy: 0.0800 - val_loss: 2.3026 - val_accuracy: 0.1000
Epoch 7/10
24/24 [==============================] - 15s 634ms/step - loss: 2.3059 - accuracy: 0.1040 - val_loss: 2.3027 - val_accuracy: 0.1000
Epoch 8/10
24/24 [==============================] - 15s 623ms/step - loss: 2.3055 - accuracy: 0.0907 - val_loss: 2.3027 - val_accuracy: 0.1000
Epoch 9/10
24/24 [==============================] - 15s 631ms/step - loss: 2.3037 - accuracy: 0.0947 - val_loss: 2.3026 - val_accuracy: 0.1000
Epoch 10/10
24/24 [==============================] - 15s 621ms/step - loss: 2.3044 - accuracy: 0.0907 - val_loss: 2.3026 - val_accuracy: 0.1000
```

**Comment by Srikeshram** [comment]
Tweaking the Learning Rate or Changing the optimizer might help you...I think this is caused due to the mismatch dataset with the optimizer....

**Comment by mrdbourke** [comment]
> Tweaking the Learning Rate or Changing the optimizer might help you...I think this is caused due to the mismatch dataset with the optimizer....

I've tried several different optimizers/learning rates but for some reason each model never moves off guessing level performance, e.g. 10% accuracy with 10 classes

**Comment by Saduf2019** [comment]
I am able to replicate the issue reported, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/38619eec760a7d0ceedebf8da8410c3b/untitled383.ipynb). Thanks!

**Comment by optiluca** [comment]
I'm also experiencing this exact same issue, was about to open a ticket myself... :-)

**Comment by optiluca** [comment]
FYI, on my side it was an input scaling issue - and looking at your gist this may be your case too!  EfficientNet is "weird" in that it simply wants untouched 0-255 inputs, while you've scaled them 0-1.

**Comment by mrdbourke** [comment]
> FYI, on my side it was an input scaling issue - and looking at your gist this may be your case too! EfficientNet is "weird" in that it simply wants untouched 0-255 inputs, while you've scaled them 0-1.

You're completely right. It turns out EfficientNet has normalization built-in.

Thank you for this.

Also found this information buried in this guide: https://keras.io/examples/vision/image_classification_efficientnet_fine_tuning/

```
from tensorflow.keras.applications import EfficientNetB0
model = EfficientNetB0(weights='imagenet')
```
"This model takes input images of shape (224, 224, 3), and the input data should range [0, 255]. Normalization is included as part of the model."

**Comment by mrdbourke** [comment]
Hmm... seems the inputs for EfficientNet are different depending on whether you're using TensorFlow Hub or `tensorflow.keras.applications`. 

TensorFlow Hub: https://tfhub.dev/tensorflow/efficientnet/b0/classification/1

> For this model, the size of the input image is flexible, but it would be best to match the model training input, which is height x width = 224 x 224 pixels for this model. The input images are expected to have color values in the range [0,1], following the common image input conventions.

`tensorflow.keras.applications`: https://keras.io/examples/vision/image_classification_efficientnet_fine_tuning/

> This model takes input images of shape (224, 224, 3), and the input data should range [0, 255]. Normalization is included as part of the model.

My assumption is that they'd be the same?

**Comment by gowthamkpr** [comment]
@mrdbourke Yes you are right. The inputs are different for hub module and keras applications module and thats the cause of the behavior that you are observing.

**Comment by mrdbourke** [comment]
@gowthamkpr do you know why the inputs are different for each model?

**Comment by n3011** [comment]
Normalization was moved into the model code keeping in mind the Serving use-case. 

So for desired behaviors
**If you import model from `tf.keras.applications` : DO NOT Normalize the inputs**

**Comment by rkuo2000** [comment]
I am getting very low accuracy (7~8%) when using TF2.3 EfficientNetB7, if switch back to efn.efficentnetB7 will get much higher accuracy (>80%) with the same Kaggle notebook except changing the model portion.

**Comment by rmothukuru** [comment]
@mrdbourke,
Yes, there are differences between **`tf.keras.applications`** and the `Models` in `Tensorflow Hub`. In addition to above comments, please refer this [Stack Overflow](https://stackoverflow.com/a/60284154/11530462) issue, Answered by Tensorflow Hub Engineer, Arno. Thanks!

**Comment by rmothukuru** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
